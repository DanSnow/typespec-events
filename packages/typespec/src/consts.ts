export const PACKAGE_NAME = '@typespec-events/typespec';
