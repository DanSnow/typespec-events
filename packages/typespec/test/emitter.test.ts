import { expect, it } from 'vitest';
import { PACKAGE_NAME } from '../src/consts.js';
import { emit } from './test-host.js';

it('emit events.zod.ts with camelCase (default)', async () => {
  const results = await emit('@event("cta_clicked") model CtaClicked { title: string}');
  expect(results['events.zod.ts']).toMatchSnapshot();
});

it('emit events.zod.ts with PascalCase', async () => {
  const results = await emit('@event("cta_clicked") model CtaClicked { title: string}', {
    options: {
      [PACKAGE_NAME]: {
        schemaNamingConvention: 'PascalCase',
      },
    },
  });
  expect(results['events.zod.ts']).toMatchSnapshot();
});
