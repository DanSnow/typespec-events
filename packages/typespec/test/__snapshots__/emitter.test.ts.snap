// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`emit events.zod.ts with PascalCase 1`] = `
"import { z } from 'zod';

export const CtaClickedSchema = z.object({ title: z.string() });

export const eventSchemas = {
  "cta_clicked": CtaClickedSchema,
} as const;
"
`;

exports[`emit events.zod.ts with camelCase (default) 1`] = `
"import { z } from 'zod';

export const ctaClickedSchema = z.object({ title: z.string() });

export const eventSchemas = {
  "cta_clicked": ctaClickedSchema,
} as const;
"
`;
